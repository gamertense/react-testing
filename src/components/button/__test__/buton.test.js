import React from 'react'
import { create, act } from 'react-test-renderer'

import Button from '../button'

describe('Button component', () => {
  test('it shows the expected text when clicked', () => {
    let component
    act(() => {
      component = create(<Button text="SUBSCRIBE TO BASIC" />)
    })
    const instance = component.root
    const button = instance.findByType('button')
    act(() => button.props.onClick())
    expect(button.props.children).toBe('PROCEED TO CHECKOUT')
  })
})

test('Matches the snapshot', () => {
  const button = create(<Button />)
  expect(button.toJSON()).toMatchSnapshot()
})
