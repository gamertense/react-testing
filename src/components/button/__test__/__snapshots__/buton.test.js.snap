// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<button
  data-testid="button"
  onClick={[Function]}
/>
`;
